chapters/en/slides/module6/module6_02.Rmd

```diff
@@ -0,0 +1,352 @@
+---
+params:
+  dynamictitle: "module6_02"
+title: "`r params$dynamictitle`"
+output:
+  md_document:
+    variant: gfm
+    pandoc_args: "--markdown-headings=atx"
+---
+
+```{r setup, include=FALSE}
+source('../../../../slide-setup.R')
+# Remember to also manually update the YAML title above
+```
+
+type: slides
+
+
+# Visualizing Geographic Data
+
+Notes:
+In this module we will see how we can plot maps with Altair,
+both as single color backgrounds
+and by linking values to different countries
+and states on the maps.
+
+---
+
+## Geographical file formats
+
+```json
+// This truncated example does not contain all border coordinates for Colorado
+{
+  "type": "Feature",
+  "id": 8,
+  "properties": {"name": "Colorado"},
+  "geometry": {
+    "type": "Polygon",
+    "coordinates": [
+      [[-106.32056285448942,40.998675790862656],[-106.19134826714341,40.99813863734313],
+       [-105.27607827344248,40.99813863734313],[-104.9422739227986,40.99813863734313],
+       [-104.05212898774828,41.00136155846029],[-103.57475287338661,41.00189871197981],
+       [-106.32056285448942,40.998675790862656]]
+    ]
+  }
+}
+```
+
+Notes:
+Up until now,
+all the data we have plotted in Altair have been read from CSV-file
+as tidy dataframes.
+To visualize data on maps we will be using
+file formats that are designed specifically for geographic data.
+
+These are called GeoJSON and TopoJSON
+and are saved as plain text file,
+but with rules for how store geographic data such as coordinates.
+You can see an example of how they look in this slide.
+
+---
+
+## Creating a map in Altair
+
+```{python}
+import altair as alt
+from vega_datasets import data
+
+world_map = alt.topo_feature(data.world_110m.url, 'countries')
+alt.Chart(world_map).mark_geoshape()
+```
+
+Notes:
+The vega sample data sets contain a TopoJSON dataset for the world and the US,
+so let's use that here.
+
+Altair work directly with GeoJSON files,
+so to use the TopoJSON sample data,
+we use the `alt.topo_feature` help function
+so that Altair can read the TopoJSON file correctly.
+
+We then use `mark_geoshape` to visualize this geographic TopoJSON data
+in the form of a map.
+Since the data contains the border coordinates of each country,
+we can see these drawn as white lines in the chart on this slide.
+
+But why does Antarctica look so much bigger
+than the rest of the world?
```

Comment:
```suggestion
The vega sample data sets contains data sets we can use to create
maps for the world and the US. There are stored in the TopoJSON format. 
We will now demonstrate how we can use such data with Altair.

To use the TopoJSON data,
we need the `alt.topo_feature` helper function
to allow Altair to correctly parse the TopoJSON file. We then use 
`mark_geoshape` to visualize this geographic TopoJSON data
in the form of a map. 

Because, the data contains the border coordinates of each country,
we can see these drawn as white lines in the chart on this slide.

Looking at this map however, you might wonder why Antarctica 
looks so much bigger than the rest of the world?
```